### PR TITLE
Replace usages of FProperty with GDK_PROPERTY that expands to correct type

### PR DESCRIPTION
**Pull request by improbable-valentyn**
#### Description
In 4.25, `UProperty` was renamed to `FProperty`. To support both, we can use a macro that expands to `USomethingProperty` or `FSomethingProperty` depending on the engine version.

#### Primary reviewers
@m-samiec @joshuahuburn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/Utils/TransientUObjectEditor.cpp [hunk 1]
```diff
@@ -21,15 +22,18 @@ namespace
...
 	{
-		const UProperty& Property = PropertyAndParent.Property;
+		const GDK_PROPERTY(Property)& Property = PropertyAndParent.Property;
 
 		if (bHaveTemplate)
 		{
+#if ENGINE_MINOR_VERSION <= 24
 			const UClass* PropertyOwnerClass = Cast<const UClass>(Property.GetOuter());
+#else
+			const UClass* PropertyOwnerClass = Property.GetOwner<const UClass>();
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/Utils/TransientUObjectEditor.cpp:32]: Isn't the outer different to the owner?

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKEditor/Private/Utils/TransientUObjectEditor.cpp:32]: I think this is taken from the updated version of the function this is copied from in the Engine.

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDKEditor/Private/Utils/TransientUObjectEditor.cpp:32]: `GetOuter` is for `UObjects`. In 4.25, `FProperty` is not a `UObject` and instead it has `GetOwner` which does pretty much the same thing.

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDKEditor/Private/Utils/TransientUObjectEditor.cpp:32]: @joshuahuburn I think you may be thinking about the UActorComponent's `GetOwner` which is different and unrelated to this.

**Pull request merged**